src/bin/julialauncher.rs

```diff
@@ -211,18 +216,36 @@ fn run_app() -> Result<i32> {
     let versiondb_data = load_versions_db(&paths)
         .with_context(|| "The Julia launcher failed to load a versions db.")?;
 
-    let mut julia_channel_to_use = config_file.data.default.clone();
-
+    // Parse command line
+    let mut channel_from_cmd_line: Option<String> = None;
     let args: Vec<String> = std::env::args().collect();
-
-    let mut julia_version_from_cmd_line = false;
-
     if args.len() > 1 {
         let first_arg = &args[1];
 
         if let Some(stripped) = first_arg.strip_prefix('+') {
-            julia_channel_to_use = Some(stripped.to_string());
-            julia_version_from_cmd_line = true;
+            channel_from_cmd_line = Some(stripped.to_string());
+        }
+    }
+
+    let julia_channel_to_use: Option<String>;
+    let juliaup_channel_source: JuliaupChannelSource;
```

Comment:
It's probably slightly more idiomatic to write the code this way:

```rust
    let julia_channel_to_use: Option<String>;
    let juliaup_channel_source: JuliaupChannelSource;

    // First check whether a channel is specified at the command line
    if channel_from_cmd_line.is_some() {
        julia_channel_to_use = channel_from_cmd_line;
        juliaup_channel_source = JuliaupChannelSource::CmdLine;
    } else {
        // Second check the JULIAUP_CHANNEL env var
        match std::env::var("JULIAUP_CHANNEL") {
            Ok(val) => {
                julia_channel_to_use = Some(val);
                juliaup_channel_source = JuliaupChannelSource::EnvVar;
            }
            Err(_) => {
                // Third use the default channel from the config file
                julia_channel_to_use = config_file.data.default.clone();
                juliaup_channel_source = JuliaupChannelSource::Default;
            }
        }
    }

    let julia_channel_to_use = julia_channel_to_use.ok_or_else(|| {
        anyhow!("The Julia launcher failed to figure out which juliaup channel to use.")
    })?;
```

And write it like this instead:

```rust
    let (julia_channel_to_use, juliaup_channel_source) = if let Some(channel) = channel_from_cmd_line {
        (channel, JuliaupChannelSource::CmdLine)
    } else if let Ok(channel) = std::env::var("JULIAUP_CHANNEL") {
        (channel, JuliaupChannelSource::EnvVar)
    } else if let Some(channel) = config_file.data.default.clone() {
        (channel, JuliaupChannelSource::Default)
    } else {
        return Err(anyhow!("The Julia launcher failed to figure out which juliaup channel to use."));
    };
```